[soundtouch] Build error on wasm32-emscripten

Package: soundtouch:wasm32-emscripten@2.3.2#2

**Host Environment**

- Host: arm64-osx
- Compiler: Clang 19.0.0
-    vcpkg-tool version: 2024-01-11-710a3116bbd615864eef5f9010af178034cb9b44
    vcpkg-scripts version: 215a25355 2024-03-01 (22 hours ago)

**To Reproduce**

`vcpkg install soundtouch`

**Failure logs**

```
-- Using cached soundtouch-soundtouch-2.3.2.tar.gz.
-- Cleaning sources at <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source <path to vcpkg>/downloads/soundtouch-soundtouch-2.3.2.tar.gz
-- Using source at <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean
-- Found external ninja('1.11.1').
-- Configuring wasm32-emscripten
-- Building wasm32-emscripten-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /Applications/CMake.app/Contents/bin/cmake --build . --config Debug --target install -- -v -j11
    Working Directory: <path to vcpkg>/buildtrees/soundtouch/wasm32-emscripten-dbg
    See logs for more information:
      <path to vcpkg>/buildtrees/soundtouch/install-wasm32-emscripten-dbg-out.log

Call Stack (most recent call first):
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/arm64-osx/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/soundtouch/portfile.cmake:24 (vcpkg_cmake_install)
  scripts/ports.cmake:172 (include)



```
<details><summary>.../vcpkg/buildtrees/soundtouch/install-wasm32-emscripten-dbg-out.log</summary>

```
Change Dir: '<path to vcpkg>/buildtrees/soundtouch/wasm32-emscripten-dbg'

Run Build Command(s): /opt/homebrew/bin/ninja -v -v -j11 install
[1/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/RateTransposer.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/RateTransposer.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/RateTransposer.cpp
em++: error: invalid optimization level: -Ofast
[2/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateCubic.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateCubic.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateCubic.cpp
em++: error: invalid optimization level: -Ofast
[3/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateShannon.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateShannon.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateShannon.cpp
em++: error: invalid optimization level: -Ofast
[4/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateLinear.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/InterpolateLinear.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/InterpolateLinear.cpp
em++: error: invalid optimization level: -Ofast
[5/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/FIFOSampleBuffer.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/FIFOSampleBuffer.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/FIFOSampleBuffer.cpp
em++: error: invalid optimization level: -Ofast
[6/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/PeakFinder.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/PeakFinder.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/PeakFinder.cpp
em++: error: invalid optimization level: -Ofast
[7/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/FIRFilter.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/FIRFilter.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/FIRFilter.cpp
em++: error: invalid optimization level: -Ofast
[8/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/AAFilter.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/AAFilter.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/AAFilter.cpp
em++: error: invalid optimization level: -Ofast
[9/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/cpu_detect_x86.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/cpu_detect_x86.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/cpu_detect_x86.cpp
em++: error: invalid optimization level: -Ofast
[10/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/mmx_optimized.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/mmx_optimized.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/mmx_optimized.cpp
em++: error: invalid optimization level: -Ofast
[11/16] /opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/BPMDetect.cpp
FAILED: CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o 
/opt/homebrew/Cellar/emscripten/3.1.54/libexec/em++ -DSOUNDTOUCH_FLOAT_SAMPLES -I<path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/include -g -Ofast -MD -MT CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o -MF CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o.d -o CMakeFiles/SoundTouch.dir/source/SoundTouch/BPMDetect.cpp.o -c <path to vcpkg>/buildtrees/soundtouch/src/2.3.2-1e000f1e3a.clean/source/SoundTouch/BPMDetect.cpp
em++: error: invalid optimization level: -Ofast
ninja: build stopped: subcommand failed.
```
</details>

--- a/ports/soundtouch/portfile.cmake
+++ b/ports/soundtouch/portfile.cmake
@@ -5,6 +5,8 @@ vcpkg_from_github(
     REF ${VERSION}
     SHA512 93f757b2c1abe16be589e0d191e6c0416c5980843bd416cd5cb820b65a705d98081c0fc7ca0d9880af54b5343318262c77ba39a096bb240ceec084e93ceef964
     HEAD_REF master
+    PATCHES
+        use-o3-emscripten.patch # Upstream PR: https://codeberg.org/soundtouch/soundtouch/pulls/29
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/soundtouch/use-o3-emscripten.patch
+++ b/ports/soundtouch/use-o3-emscripten.patch
@@ -1,0 +1,34 @@
+From 405c4586d4556982fd5bbddf1c70bc4815465c51 Mon Sep 17 00:00:00 2001
+Date: Sat, 2 Mar 2024 23:02:06 +0100
+Subject: [PATCH] Use -O3 instead of -Ofast when targeting Emscripten (WASM)
+
+---
+ CMakeLists.txt | 9 +++++++--
+ 1 file changed, 7 insertions(+), 2 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 14f23e8..dabcb9c 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -3,11 +3,16 @@ project(SoundTouch VERSION 2.3.2 LANGUAGES CXX)
+ 
+ include(GNUInstallDirs)
+ 
++set(COMPILE_OPTIONS)
++
+ if(MSVC)
+   set(COMPILE_DEFINITIONS /O2 /fp:fast)
+-  set(COMPILE_OPTIONS )
+ else()
+-  set(COMPILE_OPTIONS -Ofast)
++  if(EMSCRIPTEN)
++    list(APPEND COMPILE_OPTIONS -O3)
++  else()
++    list(APPEND COMPILE_OPTIONS -Ofast)
++  endif()
+ endif()
+ 
+ #####################
+-- 
+2.43.0
+

--- a/ports/soundtouch/vcpkg.json
+++ b/ports/soundtouch/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "soundtouch",
   "version": "2.3.2",
-  "port-version": 2,
+  "port-version": 3,
   "description": "SoundTouch is an open-source audio processing library for changing the Tempo, Pitch and Playback Rates of audio streams or audio files.",
   "homepage": "https://www.surina.net/soundtouch",
   "supports": "!uwp",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -8166,7 +8166,7 @@
     },
     "soundtouch": {
       "baseline": "2.3.2",
-      "port-version": 2
+      "port-version": 3
     },
     "soxr": {
       "baseline": "0.1.3",

--- a/versions/s-/soundtouch.json
+++ b/versions/s-/soundtouch.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "a30af1c2b397ee59f6c0048c5453ebbdf4626415",
+      "version": "2.3.2",
+      "port-version": 3
+    },
+    {
       "git-tree": "ac2649a7dd79e3a99721c357c744f81b94643a3c",
       "version": "2.3.2",
       "port-version": 2

